From 5bd94a93c89978c5e729db86b86b49919cd3b523 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?An=C3=ADbal=20Lim=C3=B3n?= <anibal.limon@linaro.org>
Date: Wed, 25 Apr 2018 12:05:29 -0500
Subject: [PATCH 5/7] main.c: Use realpath to get the actual directory of
 ptests
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix usage of relative paths in -d argument.

$ ./ptest-runner -d ./tests/data

Signed-off-by: Aníbal Limón <anibal.limon@linaro.org>
---
 main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 505829cdad58..593aff1a1956 100644
--- a/main.c
+++ b/main.c
@@ -19,6 +19,7 @@
  * 	Aníbal Limón <anibal.limon@intel.com>
  */
 
+#include <limits.h>
 #include <unistd.h>
 #include <string.h>
 #include <stdlib.h>
@@ -70,7 +71,7 @@ main(int argc, char *argv[])
 		switch (opt) {
 			case 'd':
 				free(opts.directory);
-				opts.directory = strdup(optarg);
+				opts.directory = realpath(optarg, NULL);
 				CHECK_ALLOCATION(opts.directory, 1, 1);
 			break;
 			case 'l':
-- 
2.11.0

